File: mobile/lib/services/appointment_service.dart
import '../models/appointment_model.dart';
import 'api_service.dart';

class AppointmentService {
  final _api = ApiService();

  Future<List<AppointmentModel>> getAppointments({
    String? status,
    String? date,
  }) async {
    try {
      String endpoint = '/appointments';
      List<String> params = [];
      
      if (status != null) params.add('status=$status');
      if (date != null) params.add('date=$date');
      
      if (params.isNotEmpty) {
        endpoint += '?${params.join('&')}';
      }

      final response = await _api.get(endpoint);
      
      if (response['success'] == true) {
        final appointments = (response['data']['appointments'] as List)
            .map((json) => AppointmentModel.fromJson(json))
            .toList();
        return appointments;
      }
      
      return [];
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<AppointmentModel?> createAppointment(Map<String, dynamic> data) async {
    try {
      final response = await _api.post('/appointments', data);
      
      if (response['success'] == true) {
        return AppointmentModel.fromJson(response['data']['appointment']);
      }
      
      throw Exception(response['message'] ?? 'Failed to create appointment');
    } catch (e) {
      throw Exception(e.toString());
    }
  }

  Future<bool> updateAppointmentStatus(String id, String status) async {
    try {
      final response = await _api.put('/appointments/$id/status', {
        'status': status,
      });
      
      return response['success'] == true;
    } catch (e) {
      throw Exception(e.toString());
    }
  }
}

File: web/src/services/appointmentService.js
import api from './api'
import axios from 'axios'
import { API_URL } from '../utils/constants'

export const appointmentService = {
  // Create appointment
  createAppointment: async (appointmentData) => {
    const response = await api.post('/appointments', appointmentData)
    // After interceptor: response = { success, message, data: { appointment } }
    return response.data?.appointment || response.data || response.appointment
  },

  // Get appointments
  getAppointments: async (params = {}) => {
    try {
      const queryString = new URLSearchParams(params).toString()
      const response = await api.get(`/appointments${queryString ? `?${queryString}` : ''}`)
      // Note: api interceptor returns response.data, so response IS the JSON object
      // Backend sends: { success, count, data: [...] }
      // After interceptor: response = { success, count, data: [...] }
      console.log('Appointments API response:', response)
      return response.data || response || []
    } catch (error) {
      console.error('Error fetching appointments:', error)
      throw error
    }
  },

  // Get appointment by ID
  getAppointmentById: async (id) => {
    const response = await api.get(`/appointments/${id}`)
    return response.data.appointment
  },

  // Update appointment status
  updateAppointmentStatus: async (id, status) => {
    const response = await api.put(`/appointments/${id}/status`, { status })
    return response.data.appointment
  },

  // Get available time slots
  getAvailableSlots: async (params) => {
    const queryString = new URLSearchParams(params).toString()
    const response = await api.get(`/appointments/available-slots?${queryString}`)
    return response.data
  },

  // Create anonymous appointment (no authentication required)
  createAnonymousAppointment: async (appointmentData) => {
    const response = await axios.post(`${API_URL}/appointments/anonymous`, appointmentData)
    return response.data
  },
}

